packages/components/src/AppSwitcher/AppSwitcher.component.js

```diff
@@ -0,0 +1,59 @@
+import React from 'react';
+import { useTranslation } from 'react-i18next';
+import PropTypes from 'prop-types';
+import I18N_DOMAIN_COMPONENTS from '../constants';
+
+import Action from '../Actions/Action';
+import ActionDropdown from '../Actions/ActionDropdown';
+
+import AppSwitcherCSSModule from './AppSwitcher.scss';
+import { getTheme } from '../theme';
+
+const theme = getTheme(AppSwitcherCSSModule);
+
+export default function AppSwitcher({ label, isSeparated, ...props }) {
+	const { t } = useTranslation(I18N_DOMAIN_COMPONENTS);
+	const className = theme('tc-app-switcher-action', {
+		separated: isSeparated,
+	});
+
+	let ActionComponent;
+	let clickAction;
+	let ariaLabel;
+	if (props && props.items && props.items.length) {
+		ActionComponent = ActionDropdown;
+		ariaLabel = t('APP_SWITCHER', {
+			defaultValue: 'Switch to another application. Current application: {{appName}}',
+			appName: label,
+		});
+	} else {
+		ActionComponent = Action;
+		clickAction = props.onClick;
+	}
+
+	return (
+		<li role="presentation" className={className}>
+			<span role="heading">
+				<ActionComponent
+					bsStyle="link"
+					className={theme('tc-app-switcher')}
+					tooltipPlacement="bottom"
+					label={label}
+					{...props}
+					aria-label={ariaLabel}
+					onClick={clickAction}
```

Comment:
Why don't we let the props override everything except the aria-label and onClick ?

Reply:
i was lazy. I just move the component as it was to avoid regression on this component. I look that

Reply:
hmm ok let's keep it like that to avoid any possible regressions, and if we need to override one of the two let's make another PR

Reply:
I made a change about it to apply your suggestion. I think it is ok.